eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -53,26 +53,26 @@ jobs:
         OSVmImage: 'ubuntu-18.04'
         vcpkg.deps: 'curl[ssl] paho-mqtt'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        build.args: ' -DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX'
+        build.args: ' -DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DCMAKE_BUILD_TYPE=Release'
       Win_x86_with_samples:
         OSVmImage: 'windows-2019'
         vcpkg.deps: 'curl[winssl] paho-mqtt'
         VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
-        build.args: ' -DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=WIN32'
+        build.args: ' -DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=WIN32 -DCMAKE_BUILD_TYPE=Release'
```

Comment:
You many want to consider setting this only on some build pipelines. For example .NET sets Debug for any PR validation runs that happen in our public and sets Release for any internal builds. See https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/variables/globals.yml#L8 as an example.

Reply:
Yes, that's the plan. The goal here was just exploring/brute forcing all legs to make sure they successfully build on release.

Ideally we would have simplified build configurations: https://github.com/Azure/azure-sdk-for-c/issues/639